### PR TITLE
WINDUPRULE-614 Spring data jpa to quarkus rule

### PR DESCRIPTION
**Pull request by aureamunoz**
https://issues.redhat.com/browse/WINDUPRULE-614

### CODE REVIEW

**Comment by windupgithubbot1** [comment]
Can one of the admins verify this patch?

File: rules-reviewed/quarkus/springboot/tests/springboot-data-jpa-to-quarkus.windup.test.xml [hunk 1]
```diff
@@ -0,0 +1,33 @@
+<?xml version="1.0"?>
+<ruletest id="springboot-di-to-quarkus-tests"
+          xmlns="http://windup.jboss.org/schema/jboss-ruleset"
+          xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://windup.jboss.org/schema/jboss-ruleset http://windup.jboss.org/schema/jboss-ruleset/windup-jboss-ruleset.xsd">
+    <testDataPath>data/springboot-di</testDataPath>
```
Comment (jonathanvila) [inline rules-reviewed/quarkus/springboot/tests/springboot-data-jpa-to-quarkus.windup.test.xml:5]: You should point to the folder where you are putting your artifact.
```suggestion
    <testDataPath>data/springboot-data-jpa-jpa</testDataPath>
```

File: rules-reviewed/quarkus/springboot/tests/springboot-data-jpa-to-quarkus.windup.test.xml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    <!-- By default sourceMode is True, so it will only consider .java files, not .jar binaries -->
+    <sourceMode>true</sourceMode>
```
Comment (jonathanvila) [inline rules-reviewed/quarkus/springboot/tests/springboot-data-jpa-to-quarkus.windup.test.xml:7]: Remove this as you are including a fat jar

File: rules-reviewed/quarkus/springboot/tests/springboot-data-jpa-to-quarkus.windup.test.xml [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    <testDataPath>data/springboot-data-jpa</testDataPath>
+    <!-- By default sourceMode is True, so it will only consider .java files, not .jar binaries -->
+    <sourceMode>true</sourceMode>
+    <rulePath>../springboot-data-jpa-to-quarkus.windup.xml</rulePath>
+    <ruleset>
+        <rules>
+            <rule id="springboot-data-jpa-to-quarkus-00000-test">
+                <when>
+                    <not>
+                        <hint-exists message="Replace the SpringBoot Data JPA artifact with Quarkus `spring-data-jpa` extension"/>
```
Comment (jonathanvila) [inline rules-reviewed/quarkus/springboot/tests/springboot-data-jpa-to-quarkus.windup.test.xml:14]: You are checking the title not the message.

File: rules-reviewed/quarkus/springboot/tests/springboot-data-jpa-to-quarkus.windup.test.xml [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+                    </not>
+                </when>
+                <perform>
+                    <fail message="[springboot-data-jpa-to-quarkus-extension] SpringBoot Data JPA artifact replaced by Quarkus extension was not found!" />
+                </perform>
+            </rule>
+            <rule id="springboot-data-jpa-to-quarkus-00001-test">
+                <when>
+                    <not>
+                        <hint-exists message="Replace the SpringBoot Data commons artifact with Quarkus 'spring-data-jpa' extension"/>
```
Comment (jonathanvila) [inline rules-reviewed/quarkus/springboot/tests/springboot-data-jpa-to-quarkus.windup.test.xml:24]: You are checknig the title not the message

**Comment by aureamunoz** [comment]
This can be reviewed @jonathanvila :-) Thanks for your help!

File: rules-reviewed/quarkus/springboot/springboot-data-jpa-to-quarkus.windup.xml [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+            <when>
+                <dependency groupId="org.springframework.data" artifactId="spring-data-jpa"/>
+            </when>
+            <perform>
+                <hint title="Replace the SpringBoot Data JPA artifact with Quarkus 'spring-data-jpa' extension" effort="1" category-id="mandatory">
+                    <message>
+                        Replace the SpringBoot JPA artifact with Quarkus `spring-data-jpa` extension
+
+                        Spring Data JPA is in spring-data-jpa artifact brought transitively by any `org.springframework.boot:spring-boot-*` dependency
+                        Add Quarkus dependency `io.quarkus:quarkus-spring-data-jpa-api`
```
Comment (jonathanvila) [inline rules-reviewed/quarkus/springboot/springboot-data-jpa-to-quarkus.windup.xml:27]: As I see in the documentation the dependency is :
<dependency>
    <groupId>io.quarkus</groupId>
    <artifactId>quarkus-spring-data-jpa</artifactId>
</dependency>
Not quarkus-spring-data-jpa-api

Comment (aureamunoz) [reply rules-reviewed/quarkus/springboot/springboot-data-jpa-to-quarkus.windup.xml]: that's correct! I changed it

File: rules-reviewed/quarkus/springboot/springboot-data-jpa-to-quarkus.windup.xml [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+        <rule id="springboot-jpa-to-quarkus-00000">
+            <when>
+                <dependency groupId="org.springframework.data" artifactId="spring-data-jpa"/>
+            </when>
+            <perform>
+                <hint title="Replace the SpringBoot Data JPA artifact with Quarkus 'spring-data-jpa' extension" effort="1" category-id="mandatory">
+                    <message>
+                        Replace the SpringBoot JPA artifact with Quarkus `spring-data-jpa` extension
+
+                        Spring Data JPA is in spring-data-jpa artifact brought transitively by any `org.springframework.boot:spring-boot-*` dependency
```
Comment (jonathanvila) [inline rules-reviewed/quarkus/springboot/springboot-data-jpa-to-quarkus.windup.xml:26]: Are you sure about this ?
I can see several packages with that condition `org.springframework.boot:spring-boot-*` that do not include transitively JPA  
https://mvnrepository.com/search?q=org.springframework.boot

File: rules-reviewed/quarkus/springboot/tests/springboot-data-jpa-to-quarkus.windup.test.xml [hunk 7]
```diff
@@ -0,0 +1,23 @@
+<?xml version="1.0"?>
+<ruletest id="springboot-di-to-quarkus-tests"
```
Comment (jonathanvila) [inline rules-reviewed/quarkus/springboot/tests/springboot-data-jpa-to-quarkus.windup.test.xml:2]: This test is not about DI, right ?

File: rules-reviewed/quarkus/springboot/springboot-data-jpa-to-quarkus.windup.xml (hunk 6 above)
Comment (aureamunoz) [reply rules-reviewed/quarkus/springboot/springboot-data-jpa-to-quarkus.windup.xml]: Yep, that's correct. I changed by `spring-data-*`

File: rules-reviewed/quarkus/springboot/tests/springboot-data-jpa-to-quarkus.windup.test.xml (hunk 7 above)
Comment (aureamunoz) [reply rules-reviewed/quarkus/springboot/tests/springboot-data-jpa-to-quarkus.windup.test.xml:2]: :+1:

File: rules-reviewed/quarkus/springboot/springboot-data-jpa-to-quarkus.windup.xml [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+        <rule id="springboot-jpa-to-quarkus-00000">
+            <when>
+                <dependency groupId="org.springframework.data" artifactId="spring-data-jpa"/>
+            </when>
+            <perform>
+                <hint title="Replace the SpringBoot Data JPA artifact with Quarkus 'spring-data-jpa' extension" effort="1" category-id="mandatory">
+                    <message>
+                        Replace the SpringBoot JPA artifact with Quarkus `spring-data-jpa` extension
+
+                        Spring Data JPA is in spring-data-jpa artifact brought transitively by any `org.springframework.data:spring-data-*` dependency
```
Comment (jonathanvila) [inline rules-reviewed/quarkus/springboot/springboot-data-jpa-to-quarkus.windup.xml:26]:
```suggestion
                        Spring Data JPA is in spring-data-jpa artifact brought transitively by any `org.springframework.data:spring-data-*` dependency  
```
If you want a New Line after dependency you need to add 2 spaces at the end.

Comment (aureamunoz) [reply rules-reviewed/quarkus/springboot/springboot-data-jpa-to-quarkus.windup.xml]: Ok

**Comment by jonathanvila** [comment]
LGTM :+1:

**Pull request merged**